### PR TITLE
Fix for duplicate url check bug introduced by #19734 and support utf8…

### PR DESCRIPTION
**Pull request by tonypartridge**
Fix for duplicate url check bug introduced by #19734 and support utf8 on old_urls.
Couldn't find a solution to handle this within mysql. So a simple foreach handles it perfectly fine.

@infograf768 can you please test?

### CODE REVIEW

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on dc2021a481dfadd8b4dedea4266e6e28bfed3ef6<br /><br />Great!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19950">issues.joomla.org/tracker/joomla-cms/19950</a>.</sub>

File: administrator/components/com_redirect/tables/link.php [hunk 1]
```diff
@@ -85,17 +85,20 @@ public function check()
...
 			->from('#__redirect_links')
-			->where($db->quoteName('old_url') . ' = ' . $db->quote(rawurlencode($this->old_url)));
+			->where($db->quoteName('old_url') . ' = ' . $db->quote($this->old_url));
 		$db->setQuery($query);
+		$urls = $db->loadAssocList();
 
-		$xid = (int) $db->loadResult();
-
-		if ($xid && $xid != (int) $this->id)
+		foreach ($urls AS $url)
```
Comment (Quy) [inline administrator/components/com_redirect/tables/link.php]: Change `AS` to `as`

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on dc2021a481dfadd8b4dedea4266e6e28bfed3ef6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19950">issues.joomla.org/tracker/joomla-cms/19950</a>.</sub>

**Comment by tonypartridge** [comment]
Done

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19950">issues.joomla.org/tracker/joomla-cms/19950</a>.</sub>

**Pull request merged**